### PR TITLE
Add DruidJIT with translations for most one byte bytecodes

### PR DESCRIPTION
**Pull request by guillep**
We are missing:
 - closure related bytecodes
 - write barrier related bytecodes

### CODE REVIEW

File: smalltalksrc/VMMaker/StackToRegisterMappingCogit.class.st [hunk 1]
```diff
@@ -4903,6 +4903,16 @@ StackToRegisterMappingCogit >> tryCollapseTempVectorInitializationOfSize: slots
...
 
+{ #category : #'bytecode generators' }
+StackToRegisterMappingCogit >> unknownBytecode [
+	
+	self ssFlushStack.
+	self MoveCq: bytecodePC R: ReceiverResultReg.
+	self CallRT: ceDeoptimiseFrameTrampoline.
+	deadCode := true.
+	^ 0
+]
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/StackToRegisterMappingCogit.class.st]: This is almost duplicated in Cogit class

File: smalltalksrc/VMMaker/SimpleStackBasedCogit.class.st [hunk 2]
```diff
@@ -1596,22 +1596,6 @@ SimpleStackBasedCogit >> genLoadArgAtDepth: n into: reg [
 		R: reg
 ]
 
-{ #category : #'bytecode generators' }
-SimpleStackBasedCogit >> genLongJumpIfFalse [
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/SimpleStackBasedCogit.class.st:1600]: ???

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 3]
```diff
@@ -5433,13 +5465,20 @@ StackInterpreter >> extStoreReceiverVariableBytecode [
...
+	
+	"If backwards, we should consider the delta from before the previous extensions.
+	Consider each extension is 2 bytes."
 	bcpcDelta := offset < 0 ifTrue: [numExtB * 2] ifFalse: [0].
 	extB := 0.
 	numExtB := 0.
-	instructionPointer := instructionPointer + offset.
+	self druidIgnore: [ instructionPointer := instructionPointer + offset ].
 	self ifBackwardsCheckForEvents: offset + bcpcDelta.
+	self interpreterIgnore: [ instructionPointer := instructionPointer + offset + 1 ].
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:5481]: I was thinking, after our talk about the IPs differences between compiler and interpreter.

Maybe we need a method
```st
self moveIP: offset andDo: [ self ifBackwardsCheckForEvents: offset + bcpcDelta ].
```

And override the definition for the compiler.

File: smalltalksrc/VMMaker/StackInterpreter.class.st [hunk 4]
```diff
@@ -15497,12 +15547,15 @@ StackInterpreter >> unfollow: litVar atIndex: literalIndex [
...
+
 	<option: #TempVectReadBarrier>
-	<inline: #never> "So rare it mustn't bulk up the common path"
+	<inline: #never>
 	| followed |
 	followed := objectMemory followForwarded: tempVector.
-	self itemporary: tempVectorIndex in: theFP put: followed.
-	^followed
+	self druidIgnore: [
+		self itemporary: tempVectorIndex in: theFP put: followed ].
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/StackInterpreter.class.st:15557]: 👍

**Review by PalumboN — APPROVED** [review]
RnR 👨‍🎤 🎸 ⚡

**Comment by PalumboN** [comment]
⚠️ Some related tests are failing

**Comment by guillep** [comment]
> ⚠️ Some related tests are failing

looking into this

File: smalltalksrc/VMMaker/SimpleStackBasedCogit.class.st (hunk 2 above)
Comment (guillep) [reply smalltalksrc/VMMaker/SimpleStackBasedCogit.class.st:1600]: Dead code

**Comment by guillep** [comment]
Tests are green now!

**Pull request merged**